planner/core/partition_prune.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+// // Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"github.com/pingcap/parser/model"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/sessionctx"
+	"github.com/pingcap/tidb/table"
+)
+
+// PartitionPruning finds all used partitions according to query conditions, it will
+// return nil if condition match none of partitions.
+func PartitionPruning(ctx sessionctx.Context, tbl table.PartitionedTable, conds []expression.Expression, partitionNames []model.CIStr) ([]int, error) {
```

Comment:
The return value `[]int` are hard to understand, we should at least add some comment like "The return value is a array of the idx in the partition definitions array, use pi.Definitions[idx] to get the partition ID"

Reply:
how about return a slice which contains physical table id directly?

Reply:
The most convenient return value maybe []PhysicalTable, and after that the physical table id is handy, through geting the partition name become more difficult.